### DISCUSSION TITLE
testing in nestjs without request context

### DISCUSSION THREAD

**Discussion by xenoterracide** [mikro-orm/mikro-orm#3727] [Q&A] [answered]
```
Using global EntityManager instance methods for context specific actions is disallowed.
```
how am I supposed to write a test that just tests the entity and repository, no request context, without `allowGlobalContext: true` enabled on the nestjs module that uses `rootAsync`? I don't really want to redefine that code in every single test

**Comment by xenoterracide** [comment, marked as answer]
bump!

**Reply by B4nan** [reply]
> how am I supposed to 

you are not supposed to do that. disable the context, either via ORM options or via env vars.
